fvm/programs/chain_programs.go

```diff
@@ -0,0 +1,104 @@
+package programs
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/hashicorp/golang-lru/simplelru"
+
+	"github.com/onflow/flow-go/model/flow"
+)
+
+const DefaultProgramsCacheSize = 1000
+
+// ChainPrograms is a cache of BlockPrograms databases used for speeding up
+// cadence execution.
+//
+// Since programs are derived from external source, the BlockPrograms databases
+// need not be durable and can be recreated on the fly.
+type ChainPrograms struct {
+	mutex sync.Mutex
```

Comment:
how about `RWMutex`?

Reply:
It's not safe to use RWMutex in this case since lru will modify the data structure on Get

Reply:
nice catch.